views/js/test/qtiCreator/plugins/panel/outcomeEditor/test.js

```diff
@@ -0,0 +1,126 @@
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2020 Open Assessment Technologies SA ;
+ */
+
+/**
+ * @author Anshul sharma <ansul@taotesting.com>
+ */
+define([
+    'jquery',
+    'taoQtiItem/test/qtiCreator/plugins/creatorMock',
+    'taoQtiItem/qtiCreator/plugins/panel/outcomeEditor',
+    'taoQtiItem/qtiItem/core/Loader',
+    'json!taoQtiItem/test/samples/json/airports.json'
+], function($, creatorMock, outcomeEditorPlugin, Loader, item_airport) {
+    'use strict';
+
+    QUnit.module('API');
+
+    QUnit.test('factory', function(assert) {
+        const itemCreator = creatorMock();
+
+        assert.expect(3);
+
+        assert.equal(typeof outcomeEditorPlugin, 'function', 'The module exposes a function');
+        assert.equal(typeof outcomeEditorPlugin(itemCreator), 'object', 'The factory creates an object');
+        assert.notDeepEqual(outcomeEditorPlugin(itemCreator), outcomeEditorPlugin(itemCreator), 'The factory creates an new object');
+    });
+
+    QUnit.test('plugin', function(assert) {
+        const itemCreator = creatorMock();
+        let plugin;
+
+        assert.expect(11);
+
+        plugin = outcomeEditorPlugin(itemCreator);
+
+        assert.equal(typeof plugin.init, 'function', 'The plugin has an init method');
```

Comment:
I would use a loop.

Reply:
I ll leave it like this because of lack of time, I don't see any grave downsides of this at the moment